src/core/Akka.Cluster/ClusterHeartbeat.cs

```diff
@@ -649,11 +655,12 @@ public ImmutableHashSet<UniqueAddress> Receivers(UniqueAddress sender)
                 {
                     if (iter.MoveNext() == false || n == 0)
                     {
+                        iter.Dispose(); // dispose enumerator
```

Comment:
We weren't properly disposing the enumerator from the `HeartbeatNodeRing` before.